lib/vanagon/component.rb

```diff
@@ -213,6 +224,14 @@ def configfiles
       @files.select(&:configfile?)
     end
 
+    # Retrieve all the files intended as %ghost entries for an rpm spec
+    # %files section.
+    #
+    # @return [Array] of all the rpm %ghost files.
+    def rpm_ghost_files
+      @ghost_files.to_a
```

Comment:
Why return as an Array instead of a Set?

Reply:
Well, this is mostly a tweaked variant of the Component#{config}files() methods, and while they start with an initial Set, the files() and configfiles() methods call reject and select, respectively, which force it to an Array, and the Project.get_files() expects to work with (and returns) an Array. In particular, Array#flatten() will not flatten Sets, so you get lumpy data that breaks in Project#get_rpm_ghost_files() otherwise.